manifests/init.pp

```diff
@@ -466,6 +467,7 @@
   Optional[Boolean]                       $service_hasrestart                = $docker::params::service_hasrestart,
   Optional[Variant[String,Array]]         $registry_mirror                   = $docker::params::registry_mirror,
   Boolean                                 $acknowledge_unsupported_os        = false,
+  Optional[String]                        $keyring                           = $docker::params::keyring,
```

Comment:
please use Stdlib::Absolutepath as datatype, not String. Also since there is a default value, using Optional doesn't make sense, because you cannot pass undef to the parameter:
```suggestion
  Stdlib::Absolutepath                    $keyring                           = '/etc/apt/keyrings/docker.gpg',
```